apis/core/v1beta1/function_webhook.go

```diff
@@ -17,11 +17,161 @@ limitations under the License.
 package v1beta1
 
 import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+var (
+	shipwrightStrategies           = map[string]bool{"BuildStrategy": true, "ClusterBuildStrategy": true}
+	shipwrightStrategiesSlice      = convertMapKeysToStringSlice(shipwrightStrategies)
+	funcRuntimes                   = map[Runtime]bool{Knative: true, Async: true}
+	funRuntimesSlice               = []string{"knative", "async"}
+	scaledObjectWorkloadTypes      = map[string]bool{"Deployment": true, "StatefulSet": true}
+	scaledObjectWorkloadTypesSlice = convertMapKeysToStringSlice(scaledObjectWorkloadTypes)
+	scaleJobRestartPolices         = map[v1.RestartPolicy]bool{
+		v1.RestartPolicyAlways:    true,
+		v1.RestartPolicyOnFailure: true,
+		v1.RestartPolicyNever:     true,
+	}
+	scaleJobRestartPolicesSlice = []string{"Always", "OnFailure", "Never"}
+	pubsubComponentTypes        = map[string]bool{
```

Comment:
Dapr supports [custom components](https://github.com/dapr/components-contrib). 
If we qualify the name of the component, this will cause openfunction to not support custom dapr components.

Also, I don't think the trigger type should be limited.

Reply:
@tpiperatgod @benjaminhuo This comment and proposal are inconsistent, which one should prevail?
https://github.com/OpenFunction/OpenFunction/issues/274

Reply:
We need to look for a dynamic approach to validation: depending on the version of Dapr or KEDA deployed in the cluster, then determine if the value passed into OpenFunction is within the scope of what Dapr or KEDA supports.